File: wip/serializers/task_status.py
from rest_framework import serializers

from wip.models import TaskStatus


class TaskStatusSerializer(serializers.ModelSerializer):

    class Meta:
        model = TaskStatus
        fields = [
            'id',
            'title',
            'icon',
            'colour',
            'order',
            'show_on_job_dashboard'
        ]

File: frontend/src/app/state/reducers/taskstatus.ts
import { reduceState } from '../generics';

export interface ITaskStatus {
    id: number;
    title: string;
    icon: string;
    colour: string;
    order: number;
    show_on_job_dashboard: boolean;
}

export type State = ITaskStatus[];

export const initialState: State = [];

export function reducer(state = initialState, action: any): State {
    const actionPrefix = '[TaskStatus]';
    switch (action.type) {

        // Replace objects
        case `${actionPrefix} LOAD_ALL_SUCCESS`: {
            return reduceState(state, action, 'REPLACE_ALL');
        }

        case `${actionPrefix} REPLACE_MANY`: {
            return reduceState(state, action, 'REPLACE_MANY');
        }

        // Basic CRUD actions
        case `${actionPrefix} LOAD_ONE_SUCCESS`: {
            return reduceState(state, action, 'REPLACE_ONE');
        }

        default:
            return state;
    }
}
